File: 2021/day-07/main_test.py
import unittest

import main


class TestCase(unittest.TestCase):
    def setUp(self):
        self.input = [16, 1, 2, 0, 4, 2, 7, 1, 2, 14]

    def test_part_one(self):
        self.assertEqual(main.part_one(self.input), 37)

    def test_part_two(self):
        self.assertEqual(main.part_two(self.input), 168)

File: 2022/day-01/main_test.go
package main

import "testing"

func TestPartOne(t *testing.T) {
	input, err := readLines("input_test.txt")
	if err != nil {
		panic(err)
	}

	got := partOne(input)
	want := 24000
	if got != want {
		t.Errorf("got %v, wanted %v", got, want)
	}
}

func TestPartTwo(t *testing.T) {
	input, err := readLines("input_test.txt")
	if err != nil {
		panic(err)
	}

	got := partTwo(input)
	want := 45000
	if got != want {
		t.Errorf("got %v, wanted %v", got, want)
	}
}
